### PR TITLE
Correctly set the ACE editor height

### LINKED ISSUE #7602: be_ace configuration leads to problems with longer inputs
### Affected version(s)

5.4.3

### Description

In the template be_ace (version 5.4), the line
`editor.container.style.maxHeight = ‘calc(100vh - 320px)’;`
causes a problem when entering longer code in the (ace) text-areas. The length of the display of longer entries is then limited.

@zoglo suggested removing this line as a hotfix and already discussed further solutions with @ausi 

[Slack](https://contao.slack.com/archives/CJT61P8R1/p1728051766894749?thread_ts=1728050143.399519&cid=CJT61P8R1)

### PR DESCRIPTION
**Pull request by leofeyer**
Fixes #7602

Instead of setting `editor.container.maxHeight`, the script now sets `maxLines` dynamically.

### CODE REVIEW

File: core-bundle/contao/templates/backend/be_ace.html5 [hunk 1]
```diff
@@ -44,7 +44,19 @@ if ($GLOBALS['TL_CONFIG']['useCE']):
         enableLiveAutocompletion: true
       });
 
-      editor.container.style.maxHeight = 'calc(100vh - 320px)';
+      function getMaxLines() {
+        if ('tl_files' === editor.container.closest('form').id) {
```
Comment (Toflar) [inline core-bundle/contao/templates/backend/be_ace.html5]: Looks like this should be an `eval` setting and not something hardcoded which other people cannot use?

Comment (leofeyer) [reply core-bundle/contao/templates/backend/be_ace.html5]: The feature is now triggered by adding a `data-autosize` attribute: c78dcd50df5be3ca1785c43e6f18c1fa9114ffff

**Review by fritzmg — CHANGES_REQUESTED** [review]
In my test this PR causes to always only display 14 lines (at least in Firefox 131).

**Comment by bytehead** [comment]
Seems to work for me (Firefox 131): 

Bildschirmfoto 2024-10-23 um 10 32 41

**Review by bytehead — DISMISSED** [review]

**Comment by fritzmg** [comment]
@bytehead it works for me in the template editor as well - but not in the `code` content element for example:

**Comment by bytehead** [comment]
Ah, I've missed that. Correct - in the `code` element it only shows 14 lines.

**Comment by leofeyer** [comment]
That is the intended behavior. The editor only autosizes if the `data-autosize` attribute is set and that is only the case in the file manager.

**Review by bytehead — DISMISSED** [review]

**Review by bytehead — APPROVED** [review]

**Pull request merged**